doc/isis/frr-isis-sonic-config-support-hld.md

```diff
@@ -0,0 +1,379 @@
+
+### Requirements
+
+#### Functional and Configuration Requirements
+
+  1. Extend config and management in SONiC for FRR-ISIS.
+  2. Ability to configre FRR-ISIS based on configurations from config DB with frrcfgd (sonic-buildimage/src/sonic-frr-mgmt-framework) with the field frr_mgmt_framework_config set to "true" in the DEVICE_METADATA table.
+  3. Retain access to FRR UI (vtysh) for managing FRR configs
+  4. Configure FRR-ISIS based on configurations from config DB based on defined template files.
+  5. Upon an FRR container reboot, ISIS template files will restore FRR-ISIS configs read from config DB.
+  6. Provide support for config and management of FRR-ISIS features used in SONIC with new ISIS YANG models.
+
+#### Exemptions
+Adding support for multi-linecard chassis is out of scope for this document. 
+
+### Architecture Design 
+
+There are no changes to the existing SONiC architecture. This new feature enhances existing code to include configuration support for the isisd daemon within the FRR container. Testing showed that with the isisd deamon enabled, ISIS routes are being learned directly from the FRR container without needing any changes to the existing orchagent or swss. It was observed that fpmsyncd works to push all of the ISIS learned routes from the FRR container to SONiC DB’s. 
+
+### High-Level Design 
+
+#### Design Overview
+
+This feature will extend functionality implemented in [SONiC FRR-BGP Extended Unified Configuration Management Framework](https://github.com/sonic-net/SONiC/blob/master/doc/mgmt/SONiC_Design_Doc_Unified_FRR_Mgmt_Interface.md) to support additional SONiC FRR-ISIS features. 
+
+![FRR-BGP-Unified-mgmt-frmwrk](https://user-images.githubusercontent.com/114622132/222537856-eefb1a13-bcc0-495b-938a-7ea3abee0c18.png)
+
+Diagram 1. Diagram showing the existing framework that is being extended to include support for now ISIS config schemas. This diagram is taken from and further explained in it's original feature introduction in [SONiC FRR-BGP Extended Unified Configuration Management Framework](https://github.com/sonic-net/SONiC/blob/master/doc/mgmt/SONiC_Design_Doc_Unified_FRR_Mgmt_Interface.md) to support additional SONiC FRR-ISIS features. 
+
+The Management framework will convert the YANG-based config data into requests that will write the configs into Redis DB. Redis DB events will trigger frrcfgd when the field frr_mgmt_framework_config set to "true" in the DEVICE_METADATA table, and then frrcfgd will configure FRR-ISIS using FRR CLI commands.
+
+#### Change Overview
+
+This enhancement will support FRR-ISIS features used in SONiC and all changes will reside in the sonic-buildimage repository. Changes include:
+
+- SONiC FRR-ISIS YANG models and YANG validation tests
+  - /src/sonic-yang-models
+- FRR-ISIS config template files and isisd enabled by default in the FRR container
+  - /dockers/docker-fpm-frr
+- Enable ISIS trap messages
+  - /files/image_config/copp
```

Comment:
Please capture the copp section. Please ensure trap is installed only when feature is enabled

Reply:
Based on https://github.com/sonic-net/SONiC/blob/master/doc/copp/CoPP%20Config%20and%20Management.md , it seems we should be able to add isis to the COPP_TRAP table only when needed. Would this be a sufficient solution for now ?

Reply:
Yes. Please add it on run time by updating COPP_TRAP entry. Please check is as example
```
https://github.com/sonic-net/sonic-ztp/pull/37/files
```

Reply:
Added a step to add/remove 'isis' in  the COPP_TRAP entry on bgp container startup when frr_mgmt_framework_config is updated